src/MerchantCenter/MerchantStatuses.php

```diff
@@ -237,33 +232,30 @@ protected function check_mc_is_connected() {
 	}
 
 	/**
-	 * Update stale status-related data - account issues.
+	 * Maybe start the job to refresh the status and issues data.
 	 *
 	 * @param bool $force_refresh Force refresh of all status-related data.
 	 *
+	 * @return UpdateMerchantProductStatuses The job to update the statuses.
+	 *
 	 * @throws Exception If no Merchant Center account is connected, or account status is not retrievable.
+	 * @throws NotFoundExceptionInterface  If the class is not found in the container.
+	 * @throws ContainerExceptionInterface If the container throws an exception.
 	 */
-	public function maybe_refresh_status_data( bool $force_refresh = false ): void {
-		// Only refresh if the current data has expired.
-		$this->mc_statuses = $this->container->get( TransientsInterface::class )->get( Transients::MC_STATUSES );
-		if ( ! $force_refresh && null !== $this->mc_statuses ) {
-			return;
-		}
-
-		// Save a request if accounts are not connected.
+	public function maybe_refresh_status_data( bool $force_refresh = false ): UpdateMerchantProductStatuses {
 		$this->check_mc_is_connected();
 
-		// Update account-level issues.
-		$this->refresh_account_issues();
-
-		// Update pre-sync product validation issues.
-		$this->refresh_presync_product_issues();
+		// Only refresh if the current data has expired.
+		$this->mc_statuses = $this->container->get( TransientsInterface::class )->get( Transients::MC_STATUSES );
+		$job               = $this->container->get( UpdateMerchantProductStatuses::class );
 
-		// Include any custom merchant issues.
-		$this->refresh_custom_merchant_issues();
+		// If force_refresh is true or if not transient, return empty array and scheduled the job to update the statuses.
+		if ( ! $job->is_scheduled() && ( $force_refresh || ( ! $force_refresh && null === $this->mc_statuses ) ) ) {
+			// Schedule job to update the statuses.
+			$job->schedule();
```

Comment:
Why we need double-checking if the job is running? 

AFAIK `schedule()` checks is_running() for the current page. 

I see you created the is_Schedule function without arguments to avoid page checks? 

```php
public function is_scheduled(): bool {
		// We set 'args' to null so it matches any arguments. This is because it's possible to have multiple instances of the job running with different page tokens
		return $this->is_running( null );
	}
```

Maybe we can just tweak `can_schedule` function and then rely on $job->schedule itself to check if the job is running. 

```php
/**
	 * Can the job be scheduled.
	 *
	 * @param array|null $args
	 *
	 * @return bool Returns true if the job can be scheduled.
	 */
	public function can_schedule( $args = [] ): bool {
		return parent::can_schedule( $args ) && $this->merchant_center->is_connected() && ! $this->is_running( null );
	}
```

Reply:
Thanks @puntope for the suggestion, if we do it like that, the second job won't start because at the time we schedule it, the first job hasn't finished yet.

https://github.com/woocommerce/google-listings-and-ads/blob/6b7a614b36fe71f0674025eda35e388209081d62/src/Jobs/UpdateMerchantProductStatuses.php#L101